elasticapm/transport/base.py

```diff
@@ -232,6 +232,12 @@ def _flush(self, buffer):
     def start_thread(self, pid=None):
         super(Transport, self).start_thread(pid=pid)
         if (not self._thread or self.pid != self._thread.pid) and not self._closed:
+            # Update our pid in the metadata
+            if self._metadata.get("process", {}).get("pid"):
+                self._metadata["process"]["pid"] = pid
+                if hasattr(os, "getppid"):
+                    # Windows python 2.7 was failing
+                    self._metadata["process"]["ppid"] = os.getppid()
```

Comment:
As this should only happen seldomly, we could rebuild the whole metadata object, so we can reuse the logic there (e.g. the check if `os.getppid` exists).

```suggestion
            # update metadata to reflect new process information
            if self.client:
                self._metadata = self.client._build_metadata()
```

Reply:
might wanna think about renaming `Client._build_metadata` to `Client.build_metadata` as well if we go down this route.

Reply:
Excellent suggestions. Done.